src/core/2d_primitives.js

```diff
@@ -153,8 +77,41 @@ define(function (require) {
       start = this.radians(start);
       stop = this.radians(stop);
     }
-    var curves = this._createArc(1.0, start, stop);
-    this._graphics.arc(x, y, width, height, start, stop, mode, curves);
+
+    // Adjust angles to counter linear scaling.
+    var half_pi = Math.PI / 2.0;
+    if (start >= half_pi && start < 3 * half_pi) {
+      start = Math.atan(w / h * Math.tan(start)) + 2 * half_pi;
+    } else {
+      start = Math.atan(w / h * Math.tan(start));
+    }
+    if (stop >= half_pi && stop < 3 * half_pi) {
+      stop = Math.atan(w / h * Math.tan(stop)) + 2 * half_pi;
+    } else {
+      stop = Math.atan(w / h * Math.tan(stop));
+    }
+
+    // Make all angles positive...
+    var two_pi = Math.PI * 2.0;
```

Comment:
Sure, I saw that the existing method used defined local constants so I just assumed that p5's constants were out of scope.

Can I use TWO_PI directly or do I need to do something like constants.TWO_PI?